frame/nomination-pools/src/tests.rs

```diff
@@ -1681,9 +1678,180 @@ mod unbond {
+				delegator_unbonding_eras!(3 => 6)
+			);
+			assert_eq!(
+				SubPoolsStorage::<Runtime>::get(1).unwrap(),
+				SubPools {
+					no_era: Default::default(),
+					with_era: unbonding_pools_with_era! {
+						3 => UnbondPool { points: 6, balance: 6 }
+					}
+				}
+			);
+
+			// when: casual further unbond, next era.
+			CurrentEra::set(1);
+			assert_ok!(Pools::unbond(Origin::signed(10), 10, 1));
+
+			// then
+			assert_eq!(Delegators::<Runtime>::get(10).unwrap().active_points(), 3);
+			assert_eq!(Delegators::<Runtime>::get(10).unwrap().unbonding_points(), 7);
+			assert_eq!(BondedPool::<Runtime>::get(1).unwrap().points, 3);
+			assert_eq!(
+				Delegators::<Runtime>::get(10).unwrap().unbonding_eras,
+				delegator_unbonding_eras!(3 => 6, 4 => 1)
+			);
+			assert_eq!(
+				SubPoolsStorage::<Runtime>::get(1).unwrap(),
+				SubPools {
+					no_era: Default::default(),
+					with_era: unbonding_pools_with_era! {
+						3 => UnbondPool { points: 6, balance: 6 },
+						4 => UnbondPool { points: 1, balance: 1 }
+					}
+				}
+			);
+
+			// when: unbonding more than our active: error
+			assert_noop!(
+				Pools::unbond(Origin::signed(10), 10, 5),
+				Error::<Runtime>::NotEnoughPointsToUnbond
+			);
+			// instead:
+			assert_ok!(Pools::unbond(Origin::signed(10), 10, 3));
+
+			// then
+			assert_eq!(Delegators::<Runtime>::get(10).unwrap().active_points(), 0);
+			assert_eq!(Delegators::<Runtime>::get(10).unwrap().unbonding_points(), 10);
+			assert_eq!(BondedPool::<Runtime>::get(1).unwrap().points, 0);
+			assert_eq!(
+				Delegators::<Runtime>::get(10).unwrap().unbonding_eras,
+				delegator_unbonding_eras!(3 => 6, 4 => 4)
+			);
+			assert_eq!(
+				SubPoolsStorage::<Runtime>::get(1).unwrap(),
+				SubPools {
+					no_era: Default::default(),
+					with_era: unbonding_pools_with_era! {
+						3 => UnbondPool { points: 6, balance: 6 },
+						4 => UnbondPool { points: 4, balance: 4 }
+					}
+				}
+			);
+		});
+	}
+
+	#[test]
+	fn partial_unbond_max_chunks() {
+		ExtBuilder::default().ed(1).build_and_execute(|| {
+			// so the depositor can leave, just keeps the test simpler.
+			unsafe_set_state(1, PoolState::Destroying).unwrap();
+			MaxUnbonding::set(2);
+
+			// given
+			assert_ok!(Pools::unbond(Origin::signed(10), 10, 2));
+			CurrentEra::set(1);
+			assert_ok!(Pools::unbond(Origin::signed(10), 10, 3));
+			assert_eq!(
+				Delegators::<Runtime>::get(10).unwrap().unbonding_eras,
+				delegator_unbonding_eras!(3 => 2, 4 => 3)
+			);
+
+			// when
+			CurrentEra::set(2);
+			assert_noop!(
+				Pools::unbond(Origin::signed(10), 10, 4),
+				Error::<Runtime>::MaxUnbondingLimit
+			);
+
+			// when
+			MaxUnbonding::set(3);
+			assert_ok!(Pools::unbond(Origin::signed(10), 10, 1));
+			assert_eq!(
+				Delegators::<Runtime>::get(10).unwrap().unbonding_eras,
+				delegator_unbonding_eras!(3 => 2, 4 => 3, 5 => 1)
+			);
+		})
+	}
+
+	#[test]
+	fn depositor_permissioned_partial_unbond() {
```

Comment:
A nice bonus would be to have a quick test that is just the default pool with `10`, then slash the pools active balance to 5, then have `10` try to unbond 4 which normally should now fail (but in the previous version of this PR would work)

Reply:
nice one, will do 4ef56ae4566c070dacac3290041668cd87c5d57c